examples/versatileab/src/lib.rs

```diff
@@ -0,0 +1,51 @@
+//! Common code for all examples
+
+#![no_std]
+
+// Need this to bring in the start-up function
+#[cfg(arm_profile = "a")]
+use cortex_a_rt as _;
+#[cfg(arm_profile = "r")]
+use cortex_r_rt as _;
+
+#[cfg(arm_architecture = "v8-r")]
+compile_error!("This example/board is not compatible with the ARMv8-R architecture");
+
+#[macro_export]
+macro_rules! entry_point {
+    () => {
+        /// The entry-point to the Rust application.
+        ///
+        /// It is called by the start-up code in `cortex-m-rt`.
+        #[no_mangle]
+        #[cfg(arm_profile = "r")]
+        pub extern "C" fn kmain() {
```

Comment:
Maybe we should add the CPU core argument to the cortex-R start up too?

Reply:
I did not know whether Cortex-R has multi-core. If it has, that might make sense.

Reply:
Sure.

My chip has two clusters of four R52s.

Reply:
Do you need special startup code adaption for this? For Cortex-A specifically, Both cores start executing the start up code simultaneously. The Cortex-A programmers manual recommends letting just one core do the start up routine, so I put one core to sleep with `wfe` instruction.

Reply:
I'll check the docs. But let's merge this anyway.